### PR TITLE
LPS-136630 Introduce a switch to personalise if DDM Image description field should be required

### PR DESCRIPTION
**Pull request by IstvanD**
Related Issue: https://issues.liferay.com/browse/LPS-136630

Dear Team,

Could you please review this pull request?

I made the changes that were suggested in #910.

Thank you and best regards,
István

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @IstvanD, can you please check if these failures are related to your changes? Thanks

**Comment by IstvanD** [comment]
Dear Team,

The test failures are not related to my changes.

Best regards,
István

**Comment by IstvanD** [comment]
Dear Team,

Please let me know if the solution is OK in its current form.

Thank you and best regards,
István

**Comment by rodrigopaulino** [comment]
Hi, @IstvanD.

Sorry for the response delay.
I'm still going to review your last changes, but I'm working on something else for the moment.
I'll probably get to your pull request later today or tomorrow.
I'll let you know when I restart the review.

Thanks for understanding.

**Comment by rodrigopaulino** [comment]
Hey, @IstvanD.

I'm just starting this review. I'll let you know my findings.

Thanks for your patience.

**Comment by IstvanD** [comment]
Thank you @rodrigopaulino, I'm looking forward to your findings.

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/context/DDMFormBuilderContextFactoryHelper.java [hunk 1]
```diff
@@ -262,12 +263,24 @@ protected DDMFormValues doCreateDDMFormFieldSettingContextDDMFormValues(
...
 			ddmFormFieldValue.setName(propertyName);
 
-			DDMForm ddmForm = ddmFormField.getDDMForm();
+			Object propertyValue = ddmFormFieldProperties.get(propertyName);
 
-			Value value = doCreateDDMFormFieldValue(
-				ddmFormFieldTypeSetting,
-				ddmFormFieldProperties.get(propertyName),
-				ddmForm.getAvailableLocales());
+			Value value = null;
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/context/DDMFormBuilderContextFactoryHelper.java:268]: I see that's your fix for the bug I've noticed in the previous PR. Looks good, but can you put this logic inside the `doCreateDDMFormFieldValue(DDMFormField ddmFormFieldTypeSetting, Object propertyValue, Set<Locale> availableLocales)` method? There's already some conditions to specific settings over there.

Also, this solution is only being applied to Forms, which means, Data Engine still faces the same problem as before. Please check Web Content Structures to see it happening. In order to fix that one, you'll have to include the same logic in the following method: `DataLayoutTaglibUtil._createDDMFormFieldValue(Set<Locale> availableLocales, DDMFormField ddmFormFieldTypeSetting, Object propertyValue)`.

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hi, @IstvanD. Thanks a lot for your hard work on this. It's looking great.

I've made some comments on the code. Could you please address them?

Also, I've discussed with our product manager about the need of the system setting on master and he agrees that we should remove it **only** for this version. Which means that, in upcoming versions of Liferay, the "Required Description" doesn't need to be hidden by this flag. Therefore, you don't need to verify the value of the system setting on master and you can just simply delete it from this branch. But, please don't throw away the code related to it. It will still be needed in all previous versions when you backport them, just not for our newest release.

I apologize for making these changes during your development. I hope you understand that features like this don't come in the form of bugs and we usually go over several iterations of the functionality before starting development. So this is why we're adjusting it as we review it. We're shortcutting a lot of steps of our process to make this happen.

And finally I want to thank you once again for your patience and resilience.
Your collaboration has been essential for the success of this fix.

Kind regards.

**Comment by IstvanD** [comment]
Hi @rodrigopaulino I have sent a new pull request with the suggested changes: https://github.com/liferay-forms/liferay-portal/pull/969

I'll close this one.

Best regards,
István

**Pull request closed without merging**